crates/providers/src/chain.rs

```diff
@@ -92,17 +92,26 @@ impl<P, T: Transport + Clone> ChainStreamPoller<P, T> {
             let mut retries = MAX_RETRIES;
             for number in self.next_yield..=block_number {
                 debug!(number, "fetching block");
-                let block = match provider.get_block_by_number(number, false).await {
-                    Ok(block) => block,
+                let block = match provider.get_block_by_number(number.into(), false).await {
+                    Ok(Some(block)) => block,
                     Err(RpcError::Transport(err)) if retries > 0 && err.recoverable() => {
                         debug!(number, %err, "failed to fetch block, retrying");
                         retries -= 1;
                         continue;
                     }
+                    Ok(None) if retries > 0 => {
```

Comment:
the rpc can respond with the block not existing, which can be the case if we are querying the latest block but the node doesn't have it yet. in that case i think we retry imo

Reply:
agreed